### DISCUSSION TITLE
Auto switch env files based on NODE_ENV

### DISCUSSION THREAD

**Discussion by bitsmyth** [prisma/prisma#7246] [Q&A] [answered]
I don´t think that [this](https://github.com/prisma/prisma/discussions/2310) is a proper solution for our problem. If I would implement this my whole team would have to install a global package just because we can´t detect which env file is used.

Why can´t we just extend the functionality within the `env()` function, its a one liner?

`env.${process.env.NODE_ENV}`

I tried to do it for our community but I don´t understand how the yml is working and I can´t get to the declaration of `env()`

In addition I would like to bring up an use case. I have many services running in docker which I am orchestrating with compose. As my containers are not exposed to the local machine I can´t migrate without changing DATABASE_URL manually, so multiple `env` files would be really great!

**Comment by ryands17** [comment, marked as answer]
@NikoMontana 👋 

We have this request here that you can follow that will cover this implementation: https://github.com/prisma/prisma/issues/3865

As for the workaround, a local installation will work fine. A global installation is not required and you can add `dotenv-cli` to your `devDependencies`. Then adding a simple command in the `scripts` of your `package.json` would suffice.

```json
"scripts": {
  "prisma": "dotenv -- prisma"
}
```

This would make sure that it will select the correct `.env` file based on any command you run.

As for Docker, the same implementation can be done by passing `NODE_ENV` as a variable in compose so that `dotenv` selects the correct `.env` file.
